cmd/docsync/main.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"mime"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+func main() {
+	localKeys := mustListContents(os.Args[2])
+
+	region := "us-east-1"
+	sess := session.Must(session.NewSession(aws.NewConfig().WithRegion(region)))
+	svc := s3.New(sess)
+
+	var remoteKeys []string
+	err := svc.ListObjectsPages(&s3.ListObjectsInput{
+		Bucket: aws.String(os.Args[1]),
+		Prefix: aws.String("docs/"),
+	}, func(page *s3.ListObjectsOutput, lastPage bool) bool {
+		for _, obj := range page.Contents {
+			if !(*obj.Key == "docs/") {
+				remoteKeys = append(remoteKeys, *obj.Key)
+			}
+		}
+		return true
+	})
+
+	if err != nil {
+		log.Fatalln("s3 list objects error:", err)
+	}
+
+	var prefixedLocalKeys []string
+	for _, k := range localKeys {
+		prefixedLocalKeys = append(prefixedLocalKeys, path.Join("docs", k))
+	}
+	fmt.Println("keys to upload:", len(prefixedLocalKeys)) // TEMP
+
+	remoteOnly := setDiff(remoteKeys, prefixedLocalKeys)
+	fmt.Println("keys to delete:", len(remoteOnly)) // TEMP
+
+	for _, k := range prefixedLocalKeys {
+		var body []byte
+
+		path := strings.Replace(k, "docs", os.Args[2], 1)
```

Comment:
Rather than iterate over `prefixedLocalKeys` and remove the `docs` prefix, I think it might be easier to iterate over `localKeys` and just add the prefix back when uploading to the s3 bucket.